steps/12_proba_forecasting/step.md

```diff
@@ -0,0 +1,87 @@
+
+* splitting of the prediction interval return functionality from `predict` in a function `predict_interval`, while keeping downwards compatibility
+* adding a function with distribution return type `predict_proba`
+* adding a shorthand for predictive variance return, `predict_var`
+
+## Refactor design: probabilistic forecasting
+
+We proceed outlining the refactor target interface.
+
+### Conceptual model: predict return types
+
+The conceptual idea is to map different return types onto different variants of `predict`, denoted by an underscore suffix. If we want to predict/forecast a random variable $Y$ (at a single horizon), then:
+
+* `predict` returns the predictive expectation $\mathbb{E}(Y)$
+* `predict_var` returns the predictive variance $\mbox{Var}(Y)$
+* `predict_interval` returns predictive quantiles $F_Y^{-1}(\alpha_i)$, given requested quantiles $\alpha_1,\dots, \alpha_k$ (following usual conventions on the pseudo-inverse of the cdf $F_Y$ of $Y$ rendering it well-defined)
+* `predict_proba` returns the predictive law of the random variable $Y$, i.e., a full distribution object
+
+By default, the predictive variances and distributions are point-wise (per-horizon, marginal), but the interface should also accommodates joint predictions, i.e., covariance matrix returns and joint predictive distributions.
+
+NOTE: this convention allows the `predict` return to lie outside, say, 5/95% prediction quantiles from `predict_interval`, since the return of `predict` is the predictive mean, not the predictive median; the predictive mean can, in general, be the prediction quantile at any percentage between 0 and 100.
+
+### new methods: signature specification
+
+We specify the function signature of the new methods:
+
+* `predict(self, fh=None, X=None) -> Series` is the target signature of `predict`. This differs from temporary signature to be deprecated (below).
+
+* `predict_var(self, fh=None, X=None, cov=False) -> np.ndarray` returns: if `cov=False`, an vector of same length as `fh` with predictive marginal variances; if `cov=True`, a square matrix of size `len(fh)` with predictive covariance matrix
+
+* `predict_interval(self, fh=None, X=None, alpha=Union[float, np.ndarray[float, dim=1]]) -> pd.DataFrame` returns a `pd.DataFrame` with rows having `fh` as index, and columns having name `[varname]_P[z]` where `varname` are variable names in `y` (passed to `fit`), and `z` are requested quantiles (in percentage, as float-print, without the `%` symbol). If `y` has no variable names (e.g., a `Series` without `name` attribute), the column names are `P[z]` instead. Entries are the predictive quantile for the horizon in the row, for the variable with `varname` at the quantile `z`, as in the column. If `alpha` is a float, then the returned quantiles are `alpha`, `0.5` and `1-alpha` for all variables in `y`. If `alpha` is a vector, then the returned quantiles are exactly the elements of `alpha`, for all variables in `y`.
```

Comment:
What should the return type look like for a multivariate series? Would we just add more columns or would it make sense to have a list of pd.DataFrames (one pd.DataFrame for each univariate series/variable) or some kind of nested pd.DataFrame?

Reply:
well, what I thought it should be is spelled out - just multiple columns, with the naming convention `[varname]__P[z]`. Assuming you want 90% confidence bands and pass `alpha=0.1`, and your original `y` had column names `foo`, `bar`, your return will have six columns, named `foo__P10`, `foo__P50`, `foo__P90`, `bar__P10`, `bar__P50`, `bar__P95`

Reply:
a nicer alternative could be having a column multi-index with first level being variable name, second level being a float index with values from `alpha`? That would allow to display and index non-integer percentages easily.

Perhaps that's the better way?

Reply:
I like the second option better. Seems more structured to me and it probably makes it easier to plot/process it afterwards.

Reply:
ok - I've changed the STEP to reflect this and added an example.